mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMavericksViewModel.kt

```diff
@@ -0,0 +1,191 @@
+package com.airbnb.mvrx
+
+import androidx.annotation.CallSuper
+import androidx.annotation.RestrictTo
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+import kotlin.reflect.KProperty1
+
+/**
+ * To use Mavericks, make a class MavericksViewModel or YourCompanyViewModel that extends BaseMavericksViewModel and set debugMode.
+ *
+ * Most of the time, debugMode should be set to `BuildConfig.DEBUG`. When debug mode is enabled, Mavericks will run a series of checks on
+ * your state and reducers to ensure that you are using Mavericks safely and will catch numerous common errors and mistakes.
+ *
+ * Most base classes will look like:
+ * ```
+ * abstract class MavericksViewModel<S : MvRxState>(state: S) : BaseMavericksViewModel<S>(state, debugMode = BuildConfig.DEBUG)
+ * ```
+ */
+abstract class BaseMavericksViewModel<S : MvRxState>(
+    initialState: S,
+    debugMode: Boolean,
+    /**
+     * Provide an overridden state store. This should only be used for tests and should only
+     * be exposed via a shared base class within your app. If your features extend this
+     * directly, do not override this in the primary constructor of your feature ViewModel.
```

Comment:
debug mode and store override params will be changing with the mocking framework fyi

Reply:
@elihart Yup! No problem :)